### DISCUSSION TITLE
CatBoost Classifier with cat_features

### DISCUSSION THREAD

**Discussion by ageorgiu17** [bentoml/BentoML#3479] [Q&A]
Hello! Is there a way to serve a CatBoostClassifier() that is pre-trained using cat_features? When I try to serve the model and make some predictions it raises an error

_catboost.CatBoostError: Bad value for num_feature[non_default_doc_idx=0,feature_idx=2]="Tertiary": Cannot convert 'b'Tertiary'' to float

Although the model is trained where I use object types. The input data for training the model is 

direction_1_percentage | direction_2_percentage | road_type
48.951049 | 51.048951 | Tertiary

**Comment by CristiCorches** [comment]
You can try to explicitly pass the cat_features to the bentoml runner.
It would be something like this:

```
from catboost import Pool

runner = bentoml.catboost.get("bentoml_catboost_model:latest").to_runner()

cat_features = [2] # specify your cat_features indexes
prediction = runner.predict.run(Pool(input_data, cat_features=cat_features))
```
